File: [프로그래머스] 연습문제_할인 행사.js
function solution(want, number, discount) {
    var answer = 0;
    let dic = {};
    for(let i = 0; i < want.length; i++){
        dic[want[i]] = number[i];
    }
    for(let i = 0; i <= discount.length - 10; i++){
        let check = 0;
        for(let key in dic){
            if(dic[key] > discount.slice(i, i+10).filter((el)=> el === key).length){
                check = 0;
                break;
            }
            check = 1;
        }
        answer += check;
    }
    return answer;
}

File: [프로그래머스] 연습문제_할인 행사.py
def solution(want, number, discount):
    answer = 0
    #원하는 품목이 모두 있는지 확인
    for i, j in zip(want, number):
        if j > discount.count(i):
            return 0
    for k in range(len(discount)-9):
        check = 0
        for i, j in zip(want, number):
            # k ~ k+10일 안에 원하는 품목이 존재하지 않는 경우, k+1 진행
            if j > discount[k:k+10].count(i):
                check = 0
                break
            check = 1
        answer += check
    return answer